From cbb0e683ff11d7800328da3991f3e75ef88f937f Mon Sep 17 00:00:00 2001
From: Lukas Slebodnik <lslebodn@redhat.com>
Date: Mon, 30 Jan 2017 12:49:13 +0100
Subject: [PATCH 04/79] pam_sss: Suppress warning format-truncation
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

src/sss_client/pam_sss.c: In function ‘send_and_receive’:
src/sss_client/pam_sss.c:742:39: error: ‘%.*s’ directive output
  between 0 and 18446744073709551615 bytes may cause result to exceed
  ‘INT_MAX’ [-Werror=format-truncation=]
     ret = snprintf(user_msg, bufsize, "%s%s%.*s",
                                       ^~~~~~~~~~
sssd/src/sss_client/pam_sss.c:742:39: note: assuming directive output
  of 4294967295 bytes

Reviewed-by: Fabiano Fidêncio <fidencio@redhat.com>
---
 src/sss_client/pam_sss.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/sss_client/pam_sss.c b/src/sss_client/pam_sss.c
index be697c7fcfb47a57b5b498c61f60fcf4bfbbd57f..b4175ae2c7fc1385a19f81045695bcd73d43f754 100644
--- a/src/sss_client/pam_sss.c
+++ b/src/sss_client/pam_sss.c
@@ -689,7 +689,7 @@ static int user_info_account_expired(pam_handle_t *pamh, size_t buflen,
     ret = snprintf(user_msg, bufsize, "%s%s%.*s",
                    EXP_ACC_MSG,
                    msg_len > 0 ? SRV_MSG : "",
-                   msg_len,
+                   (int)msg_len,
                    msg_len > 0 ? (char *)(buf + 2 * sizeof(uint32_t)) : "" );
     if (ret < 0 || ret > bufsize) {
         D(("snprintf failed."));
@@ -744,7 +744,7 @@ static int user_info_chpass_error(pam_handle_t *pamh, size_t buflen,
     ret = snprintf(user_msg, bufsize, "%s%s%.*s",
                    _("Password change failed. "),
                    msg_len > 0 ? _("Server message: ") : "",
-                   msg_len,
+                   (int)msg_len,
                    msg_len > 0 ? (char *)(buf + 2 * sizeof(uint32_t)) : "" );
     if (ret < 0 || ret > bufsize) {
         D(("snprintf failed."));
-- 
2.9.3

